src/utils/message_entity.rs

```diff
@@ -0,0 +1,116 @@
+use std::string::FromUtf16Error;
+
+use teloxide_core::types::MessageEntity;
+
+#[derive(Debug)]
+pub enum InvokeEntityError {
+    TooShortText { offset: usize, length: usize },
+    FromUTF16(FromUtf16Error),
+}
```

Comment:
I don't think returning an error is helpful here.
We expect that telegram returns valid entities, so we can just `panic!()`.

Reply:
i think, it's not good idea to crash the app on just little bag in telegram api.

Reply:
For the `FromUtf16Error` the bug needs to be in the `std` I think?

I just feel that the error will just be an annoyance for most users — it's rare, it's not really actionable...

Reply:
Though I guess we can have the "default" method that panics and a `try_` method that returns the `Result`, so there are options.

Reply:
And also we should probably catch panics when running handlers so panics don't shut the whole bot.

Reply:
> I just feel that the error will just be an annoyance for most users — it's rare, it's not really actionable...

ok, in that case they can just .unwrap() on their own code. In fact, they are use try operator and stay without trouble. Not panic, not crashes, correct message in log